### PR TITLE
Try to catch gubernator-breaking job changes at presubmit time

### PR DESCRIPTION
**Pull request by spiffxp**
Another attempt at fixing https://github.com/kubernetes/test-infra/issues/18770

The previous attempt at triggering the relevant tests via `bazel test //hack/...` (https://github.com/kubernetes/test-infra/pull/18448) didn't work because:
- the script was mispelled...
- ...actually bazel doesn't invoke verify-all.sh
- ...actually gubernator isn't in bazel's view of the world at all

Rather than tackle that last problem, let's just trigger the job that ends up failing more often

I won't be able to verify this as part of this PR, so look for a dummy PR to verify this post-merge

### CODE REVIEW

**Comment by spiffxp** [comment]
/wip
TIL a single [wip] commit won't mark the PR as work in progress, good to know

**Comment by BenTheElder** [comment]
> TIL a single [wip] commit won't mark the PR as work in progress, good to know

the only way a commit helps is if your PR has a single commit the title is auto-populated from the commit by default.

File: hack/verify-all.sh [hunk 1]
```diff
@@ -19,4 +19,4 @@ set -o pipefail
 
 cd "$(git rev-parse --show-toplevel)"
 find hack -name 'verify-*.sh' -not -name "$(basename "$0")" \( -print -exec '{}' ';' -o -quit \)
-./guberantor/verify_config.sh
+./gubernator/verify_config.sh
```
Comment (BenTheElder) [inline hack/verify-all.sh:22]: lol, how long has this been broken?

File: config/jobs/kubernetes/sig-testing/bazel-build-test.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ presubmits:
   kubernetes/kubernetes:
   - name: pull-kubernetes-bazel-build
     decorate: true
-    always_run: true
+    always_run: false
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-testing/bazel-build-test.yaml:5]: this seems unrelated?

Comment (spiffxp) [reply config/jobs/kubernetes/sig-testing/bazel-build-test.yaml:5]: this is a change that would require I also update gubernator's config, so this PR _should_ be failing

this is how I've discovered that bazel doesn't invoke ./hack/verify-all.sh

File: hack/verify-all.sh (hunk 1 above)
Comment (spiffxp) [reply hack/verify-all.sh]: not long, I only introduced this within the last N weeks

turns out this won't work anyway...

**Comment by chases2** [comment]
/lgtm
/hold
(For manual merging)
